### PR TITLE
[docs] Update sphinx docs and standardize docstrings

### PR DESCRIPTION
**Pull request by TurboTurtle**
This set of patches updates the Sphinx documentation setup, provides for building the docs via `setuptools` now instead of `make`, (but makes it optional to not add a `buildrequires` in the spec file), and then standardizes the docstrings for several of the larger parts of the sos API.

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -59,9 +59,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '3.2'
+version = '3.9'
 # The full version, including alpha/beta/rc tags.
-release = '3.2'
+release = '3.9.1'
```
Comment (BryanQuigley) [inline docs/conf.py:64]: did you want this to be 4.0?

Comment (TurboTurtle) [reply docs/conf.py:64]: Not until we close the release, so when we do a bump commit we'd just need to include this version here.

**Comment by BryanQuigley** [comment]
Not familiar with Sphinx - where should the generated docs end up?

As it's conditional, can we output that Sphinx is supposed to run or not?

**Comment by TurboTurtle** [comment]
The docs should eventually end up on readthedocs, however I'll need to track down who has those credentials since I think things are broken on that end as well - this PR at least fixes the build and design issues on our side.

**Comment by TurboTurtle** [comment]
> As it's conditional, can we output that Sphinx is supposed to run or not?

Added a warning when sphinx is not installed locally that docs won't be generated.

**Pull request closed without merging**